src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoHttpClient.java

```diff
@@ -248,7 +247,9 @@ private Response post(String url, String body, Map<String, String> extraCredenti
     private Response execute(final Request.Builder request, Map<String, String> extraCredentials) throws IOException {
         request.header("X-Trino-User", getUserNameForRequest());
         request.header("X-Trino-Trace-Token",tracing.currentTraceContext().get().traceIdString());
-        request.header("X-Trino-Extra-Credential", "trace=" + tracing.currentTraceContext().get().traceIdString()); // We're doing this to extract the trace token inside the SAC plugin
+        String b3Header = String.format("%s-%s-%s-%s", tracing.currentTraceContext().get().traceIdString(), tracing.currentTraceContext().get().spanIdString(),
+            1, tracing.currentTraceContext().get().parentIdString());
```

Comment:
I think the `1` here is the "sampled" flag, which you can get from `currentTraceContext().get().sampled()`. `sampled()` returns Boolean, which annoyingly can be true, false, or null. When it's null, the sampling state is "deferred" and in this case, the `b3` header should only have 3 fields.

I think you might be able to use Brave's `B3SingleFormat.writeB3SingleFormat(TraceContext)` routine to get the correct result taking [all the rules](https://github.com/openzipkin/b3-propagation#single-header) into consideration.